app/controllers/courses/subjects_controller.rb

```diff
@@ -38,6 +38,19 @@ def update
 
   private
 
+    def has_modern_languages_subject?
+      selected_subjects.any? do |s|
+        s.id.to_s == modern_languages_subject.id.to_s
+      end
+    end
+
+    def modern_languages_subject
+      return @modern_languages_subject if @modern_languages_subject
+
+      hash = @course.meta[:edit_options][:modern_languages_subject]
+      @modern_languages_subject = Subject.new(hash)
+    end
+
     def subjects_have_not_been_changed?
```

Comment:
Is `subjects_have_not_been_changed?` still in use? and `existing_non_language_subjects`

Reply:
doesn't look like it. i'll try removing

Reply:
i've removed this and a few other methods that didn't seem to be in use